library/std/src/net/addr.rs

```diff
@@ -290,7 +290,10 @@ impl SocketAddrV4 {
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn ip(&self) -> &Ipv4Addr {
-        unsafe { &*(&self.inner.sin_addr as *const c::in_addr as *const Ipv4Addr) }
+        unsafe {
+            &*(&self.inner.sin_addr as *const c::in_addr as *const super::endian::BeU32
+                as *const Ipv4Addr)
+        }
```

Comment:
Is the unsafe here still correct after this PR?
cc @RalfJung I need you here.

Reply:
I'm afraid I don't know this code or any of the involved types, and I don't have time for an `unsafe` review of the entire module -- what exactly is the question here? What is the code doing that is questionable?

Reply:
Because I changed the inner type of Ipv4Addr, I am afraid that the pointer deref through cast pointers isn't safe anymore.
I added some explanation here: https://github.com/rust-lang/rust/pull/75110#discussion_r465497588
Hope you could take a look.

Reply:
What are the involved types here? Which type is being cast to which?